### PR TITLE
Improve performance of PublicIP; add module Weather; fix compiling for Linux

### PR DESCRIPTION
**Pull request by CarterLi**
See commit list for detail

### CODE REVIEW

File: src/data/config_user.txt [hunk 1]
```diff
@@ -151,12 +151,30 @@
...
 # Sets the time to wait for the public ip server to respond.
 # Must be a positive integer.
 # Default is 0 (disabled).
 #--public-ip-timeout 0
 
+# Weather format option:
+# Sets the output weather format to be used.
+# See: https://github.com/chubin/wttr.in#one-line-output
+# Default is "%t+-+%C+(%l)".
+#--weather-format "%t+-+%C+(%l)"
```
Comment (LinusDierheimer) [inline src/data/config_user.txt:170]: This has the same name as the format string option, maybe `--weather-request`?

File: CMakeLists.txt [hunk 2]
```diff
@@ -75,7 +75,7 @@ endif()
 message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
 
 set(CMAKE_C_STANDARD 11)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wconversion")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wconversion -D_GNU_SOURCE")
```
Comment (LinusDierheimer) [inline CMakeLists.txt:78]: `target_compile_definition(libfastfetch PUBLIC _GNU_SOURCE)`

File: src/fastfetch.c [hunk 3]
```diff
@@ -1471,6 +1485,13 @@ int main(int argc, const char** argv)
     if(data.structure.length == 0)
         ffStrbufAppendS(&data.structure, FASTFETCH_DATATEXT_STRUCTURE);
 
+    if(ffStrbufContainIgnCaseS(&data.structure, ":cpuusage:"))
+        ffPrepareCPUUsage();
+    if(ffStrbufContainIgnCaseS(&data.structure, ":publicip:"))
+        ffPreparePublicIp(&instance);
+    if(ffStrbufContainIgnCaseS(&data.structure, ":weather:"))
+        ffPrepareWeather(&instance);
+
```
Comment (LinusDierheimer) [inline src/fastfetch.c:1494]: It should work without the `:` too, this way modules at the start and the end of the structure can be matched too.

**Comment by CarterLi** [comment]
Done

**Comment by LinusDierheimer** [comment]
Thanks, i will wait until the FreeBSD action works, and then merge.

**Pull request merged**